app/Http/Controllers/AuthController.php

```diff
@@ -41,33 +50,94 @@ public function register(RegisterRequest $request): JsonResponse
-            // Return an error response with the message 'Failed to register user' and HTTP status code 422 (HTTP_UNPROCESSABLE_ENTITY)
             return response()->json(['error' => 'Failed to register user'], Response::HTTP_UNPROCESSABLE_ENTITY);
         }
     }
 
+    /**
+     * Log in the user.
+     *
+     * @param LoginRequest $request The login request.
+     * @return JsonResponse The JSON response.
+     */
     public function login(LoginRequest $request): JsonResponse
     {
         $credentials = $request->validated();
-
-        if (Auth::attempt($credentials)) {
+        // Attempt to authenticate the user with the provided credentials
+        if (Auth::attempt($credentials, true)) {
+            // If authentication is successful, retrieve the authenticated user
             $user = Auth::user();
             $token = $user->createToken('authToken')->accessToken;
-
             Log::info('User logged in successfully: ' . $request->email);
-
             return response()->json(['token' => $token], Response::HTTP_OK);
         }
-
         return response()->json(['error' => 'Unauthorized'], Response::HTTP_UNAUTHORIZED);
     }
 
+    /**
+     * Reset user password.
+     *
+     * @param ResetPasswordRequest $request The reset password request.
+     * @return JsonResponse The JSON response.
+     */
+    public function resetPassword(ResetPasswordRequest $request): JsonResponse
+    {
+        $user = User::where('email', $request->input('email'))->first();
+
+        if (!$user) {
+            return response()->json(['message' => 'User not found'], Response::HTTP_NOT_FOUND);
+        }
+
+        try {
+            $resetPassword = $this->passwordService->createResetPassword($user);
+
+            Mail::to($user->email)->send(new ResetPasswordEmail($resetPassword->token));
+
+            return response()->json([
+                'message' => 'Reset password email sent'
+            ], Response::HTTP_OK);
+        } catch (Exception $e) {
+            Log::error('Error sending reset password email: ' . $e->getMessage());
+            return response()->json(['error' => 'Failed to send reset password email'],
+                Response::HTTP_UNPROCESSABLE_ENTITY);
+        }
+    }
+
+    /**
+     * Update user password.
+     *
+     * @param UpdatePasswordRequest $request The update password request.
+     * @return JsonResponse The JSON response.
+     */
+    public function updatePassword(UpdatePasswordRequest $request): JsonResponse
+    {
 
+        $resetPassword = ResetPassword::where('token', $request->token)->first();
+
+        if (!$resetPassword || !$this->passwordService->isTokenValid($resetPassword)) {
+            return response()->json(['message' => 'Invalid token, please reset password again'],
+                Response::HTTP_UNPROCESSABLE_ENTITY);
+        }
+
+        try {
+            $user = $resetPassword->user;
+
+            if (empty($request->password)) {
```

Comment:
we have already validated it in a custom request. Password always exists in request.